crates/bevy_ui/src/update.rs

```diff
@@ -131,6 +138,64 @@ fn update_clipping(
     }
 }
 
+pub fn update_ui_camera_data<T: Component>(
+    mut commands: Commands,
+    mut query: Query<
+        (
+            Entity,
+            &Camera,
+            Option<&CameraUiConfig>,
+            Option<&mut UiCameraRenderInfo>,
+            Option<ChangeTrackers<CameraUiConfig>>,
+        ),
+        (With<T>, Or<(Changed<Camera>, Changed<CameraUiConfig>)>),
+    >,
+) {
+    for (entity, camera, config, render_info, config_changed) in query.iter_mut() {
+        if matches!(config, Some(&CameraUiConfig { show_ui: false, .. })) {
+            commands.entity(entity).remove::<UiCameraRenderInfo>();
```

Comment:
The component is read-only outside of `bevy_ui`, I expose it so that it's possible to access the camera projection (for example to compute proper offset for mouse interactions)

But yeah, you are right. Between the moment the user modify `CameraUiConfig` and the next `CoreStage::PostUpdate` this component will be out of date (both in term of value and existence). Which I agree is a footgun.

I don't like this either. How would this better be expressed? It needs both access to `CameraUiConfig` and `Camera` to work, so it's not as simple as replacing the content of `CameraUiConfig`.